jupyterhub/apihandlers/users.py

```diff
@@ -330,13 +329,16 @@ async def post(self, user_name):
         if requester is None:
             # couldn't identify requester
             raise web.HTTPError(403)
+        self._jupyterhub_user = requester
+        self._resolve_scopes()
         user = self.find_user(user_name)
-        if requester is not user and not requester.admin:
-            raise web.HTTPError(403, "Only admins can request tokens for other users")
-        if not user:
-            raise web.HTTPError(404, "No such user: %s" % user_name)
-        if requester is not user:
-            kind = 'user' if isinstance(requester, User) else 'service'
+        kind = 'user' if isinstance(requester, User) else 'service'
+        scope_filter = self.get_scope_filter('users:tokens')
+        if user is None or not scope_filter(user, kind):
+            raise web.HTTPError(
+                404,
```

Comment:
Can this still be 403?

The main reason we often have 404 is to avoid leaking user info in failed authentication requests, but as long as the error code is the same for both missing and insufficient permissions, I think 403 is better, since it more naturally fits the more *likely* case of knowing the user is there, but not having the right permissions.

Reply:
Good point, thanks. I'll change it back